federation/pkg/federation-controller/sync/controller.go

```diff
@@ -490,10 +490,13 @@ func syncToClusters(clustersAccessor clustersAccessorFunc, operationsAccessor op
 		if !ok {
 			glog.Fatalf("Adapter for kind %q does not properly implement SchedulingAdapter.", kind)
 		}
-		err = schedulingAdapter.UpdateFederatedStatus(obj, schedulingInfo.Status)
-		if err != nil {
-			runtime.HandleError(fmt.Errorf("adapter.UpdateFinished() failed on adapter for %s %q: %v", kind, key, err))
-			return statusError
+		if schedulingInfo != nil {
```

Comment:
Seems odd to check for a non-nil schedulingInfo given that we have an error check above. Is a nil-defined schedulingInfo a valid state? If so, please document.